tfx/components/transform/executor.py

```diff
@@ -276,45 +277,75 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
+      exec_properties: A dict of execution properties, including:
         - module_file: The file path to a python module file, from which the
           'preprocessing_fn' function will be loaded.
         - preprocessing_fn: The module path to a python function that
           implements 'preprocessing_fn'.
+        - splits_config: A transform_pb2.SplitsConfig instance, providing splits
+          that should be analyzed and splits that should be transformed. Default
+          behavior is analyze the 'train' split (when analyze_splits is not set)
+          and transform all splits (when transform_splits is not set).
 
     Returns:
       None
     """
     self._log_startup(input_dict, output_dict, exec_properties)
 
-    train_data_uri = artifact_utils.get_split_uri(input_dict[EXAMPLES_KEY],
-                                                  'train')
-    eval_data_uri = artifact_utils.get_split_uri(input_dict[EXAMPLES_KEY],
-                                                 'eval')
+    splits_config = transform_pb2.SplitsConfig()
+    if exec_properties.get('splits_config', None):
+      json_format.Parse(exec_properties['splits_config'], splits_config)
+
+    if not splits_config.analyze_splits:
+      splits_config.analyze_splits.append('train')
+      logging.info("Analyze the 'train' split when splits_config.analyze_splits"
+                   " is not set.")
+    if not splits_config.transform_splits:
+      splits_config.transform_splits.extend(
+          artifact_utils.decode_split_names(
+              input_dict[EXAMPLES_KEY][0].split_names))
+      logging.info("Transform all splits when splits_config.transform_splits "
+                   "is not set.")
+
     schema_file = io_utils.get_only_uri_in_dir(
         artifact_utils.get_single_uri(input_dict[SCHEMA_KEY]))
     transform_output = artifact_utils.get_single_uri(
         output_dict[TRANSFORM_GRAPH_KEY])
     # TODO(b/161490287): move the split_names setting to executor for all
     #                    components.
-    transformed_example_artifact = artifact_utils.get_single_instance(
+    transform_example_artifact = artifact_utils.get_single_instance(
         output_dict[TRANSFORMED_EXAMPLES_KEY])
-    transformed_example_artifact.split_names = artifact_utils.encode_split_names(
-        artifact.DEFAULT_EXAMPLE_SPLITS)
-    transformed_train_output = artifact_utils.get_split_uri(
-        output_dict[TRANSFORMED_EXAMPLES_KEY], 'train')
-    transformed_eval_output = artifact_utils.get_split_uri(
-        output_dict[TRANSFORMED_EXAMPLES_KEY], 'eval')
+    transform_example_artifact.split_names = artifact_utils.encode_split_names(
+        splits_config.transform_splits)
+
+    analyze = []
+    for split in splits_config.analyze_splits:
+      data_uri = artifact_utils.get_split_uri(input_dict[EXAMPLES_KEY], split)
+      analyze.append(io_utils.all_files_pattern(data_uri))
+
+    transform = []
+    transformed_examples = []
```

Comment:
align with TRANSFORM_MATERIALIZE_OUTPUT_PATHS?

check with Jeffrey, he is going to update transform to support mutiple artifacts, so he probably will update the name too

Reply:
my first draft of proposed executor changes are here, for reference:

https://github.com/18jeffreyma/tfx/commit/5ab8e5975371c9537f50058cce6746a47432f301

Reply:
got it, I will check it, thanks!!